tests/code/test_spreadsheet.py

```diff
@@ -0,0 +1,194 @@
+import pytest
+
+from app.models import mainDB
+from app.models.user import User
+from app.models.term import Term
+from app.models.eventParticipant import EventParticipant
+
+from app.logic.spreadsheet import *
+
+
+@pytest.mark.integration
+def test_create_spreadsheet():
+    create_spreadsheet("2020-2021")
+
+@pytest.mark.integration
+def test_calculateRetentionRate():
+    fall_dict=({'Adopt-a-Grandparent': ['khatts'], 'CELTS-Sponsored Event': [None]})
+    spring_dict=({'Hunger Initiatives': ['neillz', 'khatts', 'ayisie', 'partont']})
+    assert calculateRetentionRate(fall_dict,spring_dict) == {'Adopt-a-Grandparent': 0.0, 'CELTS-Sponsored Event': 0.0}
+
+    fall_dict=({'Hunger Initiatives': ['neillz', 'khatts', 'ayisie', 'partont']})
+    assert calculateRetentionRate(fall_dict,spring_dict) == {'Hunger Initiatives': 1.0}
+
+    fall_dict=({'Hunger Initiatives': ['neillz', 'khatts', 'ayisie', 'partont']})
+    spring_dict=({'Hunger Initiatives': ['neillz', 'khatts', 'ayisie']})
+    assert calculateRetentionRate(fall_dict,spring_dict) == {'Hunger Initiatives': 0.75}
+
+@pytest.mark.integration
+def test_removeNullParticipants():
+    test_input_list = ['khatts']
+    assert removeNullParticipants(test_input_list) == ['khatts']
+    test_input_list = ['khatts', '', 'ayisie']
+    assert removeNullParticipants(test_input_list) == ['khatts', 'ayisie']
+
+@pytest.mark.integration
+def test_termParticipation():
+    with mainDB.atomic() as transaction:    
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts'], 'CELTS-Sponsored Event': [None]}
+
+        EventParticipant.create(user = 'partont',
+                                event = 10,
+                                hoursEarned = 1)
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts', 'partont'], 'CELTS-Sponsored Event': [None]}
+
+        EventParticipant.create(user = 'ayisie',
+                                event = 14,
+                                hoursEarned = 1)
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts', 'partont'], 'CELTS-Sponsored Event': ['ayisie']}
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_getRetentionRate():
+    with mainDB.atomic() as transaction:
+        assert getRetentionRate("2020-2021") == [('Adopt-a-Grandparent', '0.0%'), ('CELTS-Sponsored Event', '0.0%')]
```

Comment:
It looks like this working okay, but do you think we could make it so that we are manually adding someone to a fall term, and then manually adding that same person to a program event that next term. Not necessarily mandatory but I feel it may be best to do this so it is more clear what the test is doing and so we are not relying just on test data.

Reply:
This has been addressed